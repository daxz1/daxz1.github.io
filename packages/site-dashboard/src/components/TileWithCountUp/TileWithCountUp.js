import React, { memo } from "react";
import CountUp from "react-countup";

/**
 *
 * @param header
 * @param title
 * @param content
 * @param cssClasses
 * @returns {JSX.Element}
 * @constructor
 */
export const TileWithCountUp = memo(({header, title, content, cssClasses}) => (
  <div className="tile">
    <>
      <div className="tress__header">
        {header}
      </div>
      <div className="tress__title">
        {title}
      </div>
      <div className={['tress__content', cssClasses].join(' ')}>
        {content && content.value && content.suffix &&
          <CountUp
            start={0}
            end={content.value}
            suffix={content.suffix}
          />
        }
      </div>
    </>
  </div>
));
