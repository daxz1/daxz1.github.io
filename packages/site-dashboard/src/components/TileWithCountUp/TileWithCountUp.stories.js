import React from "react";
import { TileWithCountUp } from "./TileWithCountUp";

export default {
  title: "components/TileWithCountUp",
  component: TileWithCountUp,
  argTypes: {
    header: { control: "text" },
    title: { control: "text" },
    content: { control: "object" },
    cssClasses: {
      control: {
        type: "select",
        options: [
          "tress__content--sushi",
          "tress__content--flame",
          "tress__content--westside",
          "tress__content--java",
        ],
      },
    },
  },
};

const Template = (args) => <TileWithCountUp {...args} />;

export const DefaultView = Template.bind({});

DefaultView.args = {
  header: "Header Text",
  title: "Title Text",
  content: {
    value: 20,
    suffix: "%",
  },
  cssClasses: "tress__content--sushi",
};
