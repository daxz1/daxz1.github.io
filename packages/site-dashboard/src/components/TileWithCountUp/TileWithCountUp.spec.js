/*global describe, test */

import React from "react";
import { create } from "react-test-renderer";
import { TileWithCountUp } from "./TileWithCountUp";

describe("TileWithCountUp Component", () => {
  test("Matches the snapshot", () => {
    const Application = create(
      <TileWithCountUp
        header="Sales"
        title="Week on Week"
        content={{
          value: 20,
          suffix: "%",
        }}
        cssClasses={["tress__content--java"]}
      />
    );
    expect(Application.toJSON()).toMatchSnapshot();
  });
});
