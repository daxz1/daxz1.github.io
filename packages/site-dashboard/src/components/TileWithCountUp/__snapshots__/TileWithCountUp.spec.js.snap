// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileWithCountUp Component Matches the snapshot 1`] = `
<div
  className="tile"
>
  <div
    className="tress__header"
  >
    Sales
  </div>
  <div
    className="tress__title"
  >
    Week on Week
  </div>
  <div
    className="tress__content tress__content--java"
  >
    <span />
  </div>
</div>
`;
